Makefile

```diff
@@ -21,15 +22,37 @@ TAILSCALE_COMMIT=$(shell echo $(TAILSCALE_VERSION) | cut -d - -f 2 | cut -d t -f
 # Extract the version code from build.gradle.
 VERSIONCODE=$(lastword $(shell grep versionCode android/build.gradle))
 VERSIONCODE_PLUSONE=$(shell expr $(VERSIONCODE) + 1)
+ifeq ($(shell uname),Linux)
+	ANDROID_TOOLS_URL="https://dl.google.com/android/repository/commandlinetools-linux-9477386_latest.zip"
+	ANDROID_TOOLS_SUM="bd1aa17c7ef10066949c88dc6c9c8d536be27f992a1f3b5a584f9bd2ba5646a0  commandlinetools-linux-9477386_latest.zip"
+else
+	ANDROID_TOOLS_URL="https://dl.google.com/android/repository/commandlinetools-mac-9477386_latest.zip"
+	ANDROID_TOOLS_SUM="2072ffce4f54cdc0e6d2074d2f381e7e579b7d63e915c220b96a7db95b2900ee  commandlinetools-mac-9477386_latest.zip"
+endif
+ANDROID_SDK_PACKAGES='platforms;android-31' 'extras;android;m2repository' 'ndk;23.1.7779620' 'platform-tools' 'build-tools;33.0.2'
+
+# Attempt to find an ANDROID_SDK_ROOT / ANDROID_HOME based either from
+# preexisting environment or common locations, otherwise fall to
+# $PWD/android-sdk.
+export ANDROID_SDK_ROOT ?= $(shell find $$ANDROID_SDK_ROOT $$ANDROID_HOME $$HOME/Library/Android/sdk $$HOME/Android/Sdk /usr/lib/android-sdk -maxdepth 1 -type d 2>/dev/null | head -n 1)
+ifeq ($(ANDROID_SDK_ROOT),)
+export ANDROID_SDK_ROOT=$(PWD)/android-sdk
+endif
+export ANDROID_HOME ?= $(ANDROID_SDK_ROOT)
+
+# Set JAVA_HOME to the Android Studio bundled JDK.
+# TODO(raggi): what is the equivalent location on Linux?
+export JAVA_HOME ?= $(shell find /Applications/Android\ Studio.app/Contents/jre/Contents/Home -maxdepth 1 -type d 2>/dev/null | head -n 1)
 
-TOOLCHAINREV=$(shell go run tailscale.com/cmd/printdep --go)
+TOOLCHAINREV=$(shell go run tailscale.com/cmd/printdep@$(TAILSCALE_MOD_VERSION) --go)
```

Comment:
Changes here might break the F-Droid builder. We'll have to go find it, but I recall they patch the Makefile to remove the Tailscale toolchain.

I imagine we'd proceed to do this, but if we can find where it is we might file an issue explaining what they'll run into.

Reply:
Is this comment referring to a generality of changing lines that will cause a patch hunk to fail to apply, or something specific about the change?

Reply:
I believe this PR will break the FDroid build https://gitlab.com/fdroid/fdroiddata/-/commit/e0b078c0db72c959a73caedbb14f6ad54f1a783a

However I don't see a way to avoid breaking the FDroid build. A sed done on our Makefile will break when we change the Makefile.

Therefore I think we proceed, and we help get FDroid updated once submitted.

Reply:
Yeah, agreed. I think perhaps f-droid might want a clean(er) way to override the toolchain behaviors in the makefile.

Reply:
It was possible to clean this up some more relatively easily, as the old want/check dance was not in use in the new arrangement. Now the user can simply pass a custom TOOLCHAINDIR, which must contain a `bin/go` or we'll attempt to download our preferred version.